### PR TITLE
Fix hypre_CSRMatrixAddHost

### PR DESCRIPTION
**Pull request by liruipeng**
Fix problems in #341

### CODE REVIEW

File: src/seq_mv/csr_matop.c [hunk 1]
```diff
@@ -412,24 +412,9 @@ hypre_CSRMatrixAddHost ( hypre_CSRMatrix *A,
 #endif
    {
       HYPRE_Int   ns, ne;
-      HYPRE_Int   ii, size, rest, num_threads;
       HYPRE_Int  *marker = NULL;
 
-      ii = hypre_GetThreadNum();
-      num_threads = hypre_NumActiveThreads();
-
-      size = nnzrows_A/num_threads;
```
Comment (liruipeng) [inline src/seq_mv/csr_matop.c:421]: @victorapm Note that you had `nnzrows_A` which should be `nnzrows_C` I believe. I also use the existing function `hypre_partition1D` to simplify the code.

Comment (victorapm) [reply src/seq_mv/csr_matop.c:421]: Yes, right! This looks much better and solves the issue. Should we use `hypre_partition1D` at https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L527 ; https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L737 ;
https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L1143 ;
https://github.com/hypre-space/hypre/blob/cd7cd75786646c289ab15cf86337c17249e4422d/src/seq_mv/csr_matop.c#L1372 ;

as well? Thanks!

**Review by victorapm — APPROVED** [review]
Thanks for this, Ruipeng!

File: src/seq_mv/csr_matop.c (hunk 1 above)
Comment (liruipeng) [reply src/seq_mv/csr_matop.c:421]: Yes, I prefer using that function, but what's more important is we don't have similar bugs at other places. Can you please double check all of them? Thanks!

Comment (victorapm) [reply src/seq_mv/csr_matop.c:421]: Yes! I just double checked and they are correct. Thanks!

**Comment by liruipeng** [comment]
Btw I found this problem by running `mpirun -np 2  ./maxwell_unscaled -in TEST_sstruct/sstruct.in.maxwell3d_unscaled -solver 1 -P 2 1 1` for which you also need to have 
```
   hypre_ParAMGDataRAP2(amg_data)              = 1;
   hypre_ParAMGDataModularizedMatMat(amg_data) = 1;
```
which are the default with `HYPRE_USING_GPU`. It also has the same problem running on CPUs

**Comment by victorapm** [comment]
Interesting, should we add `mpirun -np 2 ./maxwell_unscaled -in TEST_sstruct/sstruct.in.maxwell3d_unscaled -solver 1 -P 2 1 1 -rap 1 -mod_rap2 1` to the CPU tests?

File: src/seq_mv/csr_matop.c (hunk 1 above)
Comment (liruipeng) [reply src/seq_mv/csr_matop.c:421]: > Yes! I just double checked and they are correct. Thanks!

Great! Do you also want to make more changes using `hypre_partition1D`? I am fine with it or without. Thanks!

**Comment by liruipeng** [comment]
> -rap 1 -mod_rap2 1

This driver doesn't take `-rap 1 -mod_rap2 1` as cmd-line args...It would be nice to test

File: src/seq_mv/csr_matop.c (hunk 1 above)
Comment (victorapm) [reply src/seq_mv/csr_matop.c:421]: I think it's a good idea. Then we reduce code duplication. Can i do this and commit to this branch?

Comment (liruipeng) [reply src/seq_mv/csr_matop.c:421]: > I think it's a good idea. Then we reduce code duplication. Can i do this and commit to this branch?

Of course you can :)

**Comment by victorapm** [comment]
> > -rap 1 -mod_rap2 1
> 
> This driver doesn't take `-rap 1 -mod_rap2 1` as cmd-line args...It would be nice to test

Oh, that's true. At least we can still rely on the GPU tests :)

**Comment by liruipeng** [comment]
> > > -rap 1 -mod_rap2 1
> > 
> > 
> > This driver doesn't take `-rap 1 -mod_rap2 1` as cmd-line args...It would be nice to test
> 
> Oh, that's true. At least we can still rely on the GPU tests :)

Yes, for now. But eventually maxwell solvers will be ported to GPUs, then `hypre_CSRMatrixAddHost` will not be used anymore with GPU runs.

**Pull request merged**